pages/_app.tsx

```diff
@@ -1,10 +1,67 @@
 import Head from "next/head"
+import { useEffect } from "react"
+import { useRouter } from "next/router"
+import Script from "next/script"
 import "@nypl/design-system-react-components/dist/styles.css"
 
+import { trackVirtualPageView } from "../src/utils/appUtils"
+import { appConfig } from "../src/config/config"
+
 // eslint-disable-next-line @typescript-eslint/naming-convention
 function App({ Component, pageProps }) {
+  const router = useRouter()
+
+  // TODO: The code below is a verbose solution for page view tracking
+  // in Adobe Analytics that guarantees that page views will only be sent
+  // on the first app render or on Next route changes.
+  // We should determine if the simple useEffect solution is reliable enough for
+  // page view tracking.
+
+  // // Prevents double-firing of useEffect on initial page load
+  // const initialized = useRef(false)
+
+  // // Track initial page view to Adobe Analytics
+  // useEffect(() => {
+  //   if (!initialized.current) {
+  //     initialized.current = true
+  //     trackVirtualPageView(router.asPath)
+  //   }
+  // }, [router.asPath])
+  //
+  // // Track subsequent page views to Adobe Analytics
+  // useEffect(() => {
+  //   const handleRouteChange = (url: string) => {
+  //     trackVirtualPageView(url)
+  //   }
+  //   // When the component is mounted, subscribe to router changes
+  //   // and track those page views
+  //   router.events.on("routeChangeComplete", handleRouteChange)
+  //
+  //   // If the component is unmounted, unsubscribe
+  //   // from the event with the "off" method
+  //   return () => {
+  //     router.events.off("routeChangeComplete", handleRouteChange)
+  //   }
+  // }, [router.events])
+
+  // Track page view events to Adobe Analytics
+  useEffect(() => {
+    trackVirtualPageView(router.asPath)
+  })
+
   return (
     <>
+      <Script async src={appConfig.adobeEmbedUrl[appConfig.environment]} />
+      <Script id="adobeDataLayerDefinition">
```

Comment:
I noticed this was moved from _document to _app. I think scripts set and called in _document only get called once. I don't think that's true in _app so this doesn't seem right. It's one reason why the header/footer are set in _document but do verify this.